### ISSUE TITLE
Feature Request: Option for "Add Next Occurrence" to select whitespace

### ISSUE THREAD

**Issue by joncalhoun** [Microsoft/vscode#45001]
In both Atom and Sublime Text when you use the multiselect shortcut (`cmd+d` - Add Next Occurrence equivalent in VSCode) on whitespace, such as the whitespace used to indent some HTML, it will select all of the whitespace. VSCode doesn't do anything when "Add Next Occurrence" is performed on whitespace.

Would it be possible to change this via a setting, or even by making it the only behavior? As of now the Add Next Occurrence command is useless on whitespace.

Here is the desired behavior (shown in Atom):

![](https://www.dropbox.com/s/kmnnq2yr4d35hz9/vscode.gif?dl=0&raw=1)

## Additional info

- VSCode Version: 1.20.1
- OS Version: macOS High Sierra 10.13.3

Steps to Reproduce:

1. Move your cursor into empty space, such as the spaces used to indent some HTML
2. Hit `cmd+d` (Add Next Occurence). Nothing happens.

Does this issue occur when all extensions are disabled?: Yes

**Comment by joncalhoun** [comment]
If I were to create a PR for this at some point would it be accepted, or do feature requests need to go through an approval process of some sort first?

I'm assuming the PR would need a config setting, the actual feature, and tests. Am I missing anything else?

**Comment by rmacy** [comment]
Please add this!

**Comment by peterkagey** [comment]
This seems like a bug to me, not a feature request. Ordinarily, when you "Add Next Occurrence" when nothing is selected, it has the same behavior as when you double click in that position with your mouse.

This has had multiple questions asked about it on reddit.com/r/vscode: https://www.reddit.com/r/vscode/comments/8am57h/how_to_select_whitespace_between_text/
https://www.reddit.com/r/vscode/comments/foa1hr/how_to_select_whitespace_between_text/

**Comment by joncalhoun** [comment]
I agree that this feels somewhat like a bug, but I suspect it would be very hard to change now so I proposed it as a feature/option rather than a bug.

**Comment by hediet** [comment]
> This seems like a bug to me, not a feature request. Ordinarily, when you "Add Next Occurrence" when nothing is selected, it has the same behavior as when you double click in that position with your mouse.

I like that.

> If I were to create a PR for this at some point would it be accepted

I suggest to implement this as an extension first (which should be doable in less than an hour) and see how many installs this extension gets (the extension can contribute a command that overrides Cmd+D). If there is demand, we are open to accept PRs.
